src/main/kotlin/io.javalin/Javalin.kt

```diff
@@ -0,0 +1,309 @@
+        return addHandler(Handler.Type.HEAD, path, handler)
+    }
+
+    fun trace(path: String, handler: Handler): Javalin {
+        return addHandler(Handler.Type.TRACE, path, handler)
+    }
+
+    fun connect(path: String, handler: Handler): Javalin {
+        return addHandler(Handler.Type.CONNECT, path, handler)
+    }
+
+    fun options(path: String, handler: Handler): Javalin {
+        return addHandler(Handler.Type.OPTIONS, path, handler)
+    }
+
+    // Secured HTTP verbs
+    operator fun get(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.get(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun post(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.post(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun put(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.put(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun patch(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.patch(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun delete(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.delete(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun head(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.head(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun trace(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.trace(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun connect(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.connect(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    fun options(path: String, handler: Handler, permittedRoles: List<Role>): Javalin {
+        return this.options(path, Handler { req, res -> accessManager.manage(handler, req, res, permittedRoles) })
+    }
+
+    // Filters
+    fun before(path: String, handler: Handler): Javalin {
+        return addHandler(Handler.Type.BEFORE, path, handler)
+    }
+
+    fun before(handler: Handler): Javalin {
+        return before("/*", handler)
+    }
+
+    fun after(path: String, handler: Handler): Javalin {
+        return addHandler(Handler.Type.AFTER, path, handler)
+    }
+
+    fun after(handler: Handler): Javalin {
+        return after("/*", handler)
+    }
+
+    // Reverse routing
+    fun pathFinder(handler: Handler): String? {
+        return pathMatcher.findHandlerPath({ he -> he.handler == handler })
+    }
```

Comment:
One liners are possible in Kotlin and method/functions that the last parameter is a function, such parameter could be put outside the parenthesis.

```kotlin
fun pathFinder(handler: Handler): String? = pathMatcher.findHandlerPath { he -> he.handler == handler }
```